kedro/io/data_catalog.py

```diff
@@ -615,15 +616,47 @@ def add_transformer(
                 )
             self._transformers[data_set_name].append(transformer)
 
-    def list(self) -> List[str]:
-        """List of ``DataSet`` names registered in the catalog.
+    def list(self, regex_search: Optional[str] = None) -> List[str]:
+        """
+        List of all ``DataSet`` names registered in the catalog.
+        This can be filtered by providing an optional regular expression
+        which will only return matching keys.
 
+        Args:
+            regex_search: An optional regular expression which can be provided
+                to limit the data sets returned by a particular pattern.
         Returns:
-            A List of ``DataSet`` names, corresponding to the entries that are
-            registered in the current catalog object.
+            A list of ``DataSet`` names available which match the
+            `regex_search` criteria (if provided). All data set names are returned
+            by default.
+
+        Raises:
+            SyntaxError: When an invalid regex filter is provided.
 
+        Example:
+        ::
+
+            >>> io = DataCatalog()
+            >>> # get data sets where the substring 'raw' is present
+            >>> raw_data = io.list(regex_search='raw')
+            >>> # get data sets which start with 'prm' or 'feat'
+            >>> feat_eng_data = io.list(regex_search='^(prm|feat)')
+            >>> # get data sets which end with 'time_series'
+            >>> models = io.list(regex_search='.+time_series$')
         """
-        return list(self._data_sets.keys())
+
+        if regex_search is None:
+            return list(self._data_sets.keys())
+
+        if not regex_search.strip():
```

Comment:
What if they give `regex_search=" "` to list all dataset names that contain a space character?

Reply:
This would work alternatively `\s` should work. I can’t remember if yaml keys can have spaces, but if so I would expect this behaviour